paddle/phi/api/generator/dist_api_gen.py

```diff
@@ -1898,6 +1925,11 @@ def generate_auto_parallel_branch(self) -> str:
         kernel_call_code = self.generate_kernel_call_code()
         fallback_code = self.generate_fallback_code()
         output_dist_attr_setting = self.generate_output_dist_attr_setting()
+        # In some cases like `nonzero`, the output shape is determined after kernel call.
+        # Therefore, we need to check and backfill global shape after kernel call.
+        check_and_backfill_global_shape_code = (
```

Comment:
generate only for the op that contains -1 shape.

Reply:
> generate only for the op that contains -1 shape.

done~